core/deployment/src/main/java/io/quarkus/deployment/dev/ClassScanResult.java

```diff
@@ -37,7 +37,7 @@ public static ClassScanResult merge(ClassScanResult m1, ClassScanResult m2) {
         ret.changedClassNames.addAll(m2.changedClassNames);
         ret.deletedClassNames.addAll(m2.deletedClassNames);
         ret.addedClassNames.addAll(m2.addedClassNames);
-        ret.compilationHappened = m1.compilationHappened | m2.compilationHappened;
+        ret.compilationHappened = m1.compilationHappened || m2.compilationHappened;
```

Comment:
these are two booleans, it's totally fine to use the single `|` ; why would you want to change this? A matter of style?

Reply:
Mainly because it's more efficient, since using the single `|` means that all expressions would need to be checked, even if the previous expressions already determine the value of the entire statement. Short-circuit operators are also more readable since they're more commonly used.

Reply:
hi @clara0 , yes that's a good point in general but it doesn't really apply when you have primitive booleans, so I'll close this PR.

But thanks for checking all cases in the project!